lib/src/dart_dev_cli.dart

```diff
@@ -101,7 +101,7 @@ Future _run(List<String> args) async {
 
   ArgResults env;
   try {
-    env = _parser.parse(args);
+    env = tryArgsCompletion(args, _parser);
```

Comment:
I didn't read through the full source, but it seems like it basically just adds a new command that provides completions. The completion script is then almost entirely generic in that it just runs a special version of ddev itself to get the completions.